desktop-widgets/printdialog.cpp

```diff
@@ -215,12 +227,25 @@ void PrintDialog::printClicked(void)
 {
 	createPrinterObj();
 	QPrintDialog printDialog(qprinter, this);
+#ifdef USE_WEBENGINE
+	connect(printer, SIGNAL(progessUpdated(int)), progressBar, SLOT(setValue(int)));
+	connect(printer, &Printer::jobDone, this, &PrintDialog::printingDone);
+	printer->print();
+}
+
+void PrintDialog::printingDone()
+{
+	close();
+}
+
+#else
```

Comment:
The #ifdef formatting is clearly bad. About the connect: Will have to look into this: The thing is that rendering in WebEngine in not synchronous, so you have to listen to the completed signal before cleaning things up. But together with the preview, this might need more consideration.